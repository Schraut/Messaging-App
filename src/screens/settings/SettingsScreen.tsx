import * as React from "react";
import { Text, View, StyleSheet, SafeAreaView, Image } from "react-native";
import { Theme, useTheme } from "../../../resources/theme/theme";
import SettingsButton from "../../components/buttons/SettingsButton";
import Icon from "../../components/icon/Icon";
import SettingsHeader from "./SettingsHeader";

interface SettingsScreenProps {
  navigation: any; //we may not need this. Depends on if we take it out of bottomtabs
  firstName: string;
  lastName: string;
  username: string;
  profileImage: string;
}

export default function SettingsScreen({
  navigation,
  firstName,
  lastName,
  username,
  profileImage,
}: SettingsScreenProps) {
  const theme = useTheme();
  const styles = stylesHandler(theme);

  return (
    <SafeAreaView style={styles.container}>
      <SettingsHeader navigation={navigation} />
      <View style={styles.accountInfo}>
        <Image source={{ uri: profileImage }} style={styles.profileImage} />
        <Text style={styles.nameText}>
          {firstName} {lastName}
        </Text>
        <Text style={styles.usernameText}>{username}</Text>
      </View>
      <SettingsButton
        onPress={() => {}}
        text="Account"
        icon={
          <Icon
            iconName="settings"
            size={24}
            iconType="MaterialIcon"
            color={theme.colors.inverseBackground}
          />
        }
      />
      <SettingsButton
        onPress={() => {}}
        text="Notifications"
        icon={
          <Icon
            iconName="settings"
            size={24}
            iconType="MaterialIcon"
            color={theme.colors.inverseBackground}
          />
        }
      />
      <SettingsButton
        onPress={() => {}}
        text="Privacy Policy"
        icon={
          <Icon
            iconName="settings"
            size={24}
            iconType="MaterialIcon"
            color={theme.colors.inverseBackground}
          />
        }
      />
      <SettingsButton
        onPress={() => {}}
        text="Ratings"
        icon={
          <Icon
            iconName="settings"
            size={24}
            iconType="MaterialIcon"
            color={theme.colors.inverseBackground}
          />
        }
      />
      <Text style={styles.text}>Buttons above for testing</Text>
    </SafeAreaView>
  );
}

const stylesHandler = (theme: Theme) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.background,
      alignItems: "center",
    },
    accountInfo: {
      width: "100%",
      justifyContent: "center",
      alignItems: "center",
      paddingVertical: theme.spacing(4),
      borderBottomWidth: 1,
      borderBottomColor: theme.colors.staticGrey,
    },
    profileImage: {
      width: theme.spacing(25),
      height: theme.spacing(25),
      borderRadius: theme.spacing(25) / 2, //to get 50
    },
    nameText: {
      color: theme.colors.inverseBackground,
      fontFamily: theme.font.header.fontFamily,
      fontSize: theme.font.header.fontSize,
    },
    usernameText: {
      color: theme.colors.inverseBackground,
      fontFamily: theme.font.textPrimary.fontFamily,
      fontSize: theme.font.textPrimary.fontSize,
    },
    text: {
      color: theme.colors.inverseBackground,
      fontFamily: theme.font.textPrimary.fontFamily,
      fontSize: theme.font.textPrimary.fontSize,
      marginTop: theme.spacing(2),
    },
  });
