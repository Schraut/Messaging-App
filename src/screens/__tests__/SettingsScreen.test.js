import React from "react";
import renderer from "react-test-renderer";
import ThemeContent from "../../../resources/theme/theme";
import SettingsScreen from "../SettingsScreen";

describe("Default SettingsScreen", () => {
  it("renders in light mode", () => {
    const tree = renderer
      .create(
        <ThemeContent>
          <SettingsScreen />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders in dark mode", () => {
    const tree = renderer
      .create(
        <ThemeContent isDarkMode={true}>
          <SettingsScreen />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
