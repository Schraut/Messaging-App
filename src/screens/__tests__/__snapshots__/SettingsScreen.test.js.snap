// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default SettingsScreen renders in dark mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "System",
        "fontSize": 12,
      }
    }
  >
    Settings!
  </Text>
</View>
`;

exports[`Default SettingsScreen renders in light mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 12,
      }
    }
  >
    Settings!
  </Text>
</View>
`;
